src/library_pthread.js

```diff
@@ -451,25 +447,33 @@ var LibraryPThread = {
 
     getNewWorker: function() {
       if (PThread.unusedWorkers.length == 0) {
-#if !PROXY_TO_PTHREAD && PTHREAD_POOL_SIZE_STRICT
-#if ASSERTIONS
-        err('Tried to spawn a new thread, but the thread pool is exhausted.\n' +
-        'This might result in a deadlock unless some threads eventually exit or the code explicitly breaks out to the event loop.\n' +
-        'If you want to increase the pool size, use setting `-sPTHREAD_POOL_SIZE=...`.'
-#if PTHREAD_POOL_SIZE_STRICT == 1
-        + '\nIf you want to throw an explicit error instead of the risk of deadlocking in those cases, use setting `-sPTHREAD_POOL_SIZE_STRICT=2`.'
-#endif
-        );
-#endif // ASSERTIONS
-
-#endif
-#if !PROXY_TO_PTHREAD && PTHREAD_POOL_SIZE_STRICT == 2
-        // Don't return a Worker, which will translate into an EAGAIN error.
-        return;
-#else
-        PThread.allocateUnusedWorker();
-        PThread.loadWasmModuleToWorker(PThread.unusedWorkers[0]);
-#endif
+        // PTHREAD_POOL_SIZE_STRICT should show a warning and, if set to level `2`, return from the function.
+        // However, if we're in Node.js, then we can create new workers on the fly and PTHREAD_POOL_SIZE_STRICT
+        // should be ignored altogether.
+        #if PTHREAD_POOL_SIZE_STRICT && ASSERTIONS || PTHREAD_POOL_SIZE_STRICT == 2
```

Comment:
We don't indent pre-processor lines..  we could change that style rule but I'd rather not do that as part of this change.

Reply:
I was always wondering why btw, given that we do support that in our preprocessor. In this particular case I just got completely lost and had to indent them for my own understanding before modifying conditions 😅

I can unindent them back now if you want, sure.

As a more overarching wish, I'd really like if we instead relied on JS DCE instead or something like that instead of C-like preprocessing, to make JS files more IDE- and user-friendly.